### PR TITLE
drivers: usb: Add OTGFS USB device controller

### PR DESCRIPTION
**Pull request by lmikl**
Begin to USB for gigadevice GD32f4xx. only FS.

Only issue : Sometimes it have 2 "same" interrupts on the receive endpoint. So it receive 2 times the same message.
If someone can be search the issue.

### CODE REVIEW

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please read
- https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style
- https://kernel.org/doc/html/latest/process/coding-style.html
- https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines
- https://docs.zephyrproject.org/latest/connectivity/usb/device/api/usb_dc.html
- Please no "remove comment" or similar commits.
- There must be no doxygen comments in the driver source code.

File: dts/bindings/usb/gd,gd32-otgfs.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  pinctrl-names:
+    required: true
+
+  ram-size:
+    type: int
+    required: true
+    description: |
+      Size of USB dedicated RAM. GD32 SOC's reference
+      manual defines a shared FIFO size.
+
```
Comment (nandojve) [inline dts/bindings/usb/gd,gd32-otgfs.yaml:29]: Try use up to 80 cols always possible.
You must add devicetree documentation here with examples. Each non standard property should be very well documented in a way that devicetree maintainers don't need look in the datasheet.

File: dts/bindings/usb/gd,gd32-otgfs.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  phys:
+    type: phandle
+    description: PHY provider specifier
```
Comment (nandojve) [inline dts/bindings/usb/gd,gd32-otgfs.yaml:32]: You must add devicetree documentation here with examples. Each non standard property should be very well documented in a way that devicetree maintainers don't need look in the datasheet.

**Review by nandojve — CHANGES_REQUESTED** [review]
I took liberty to change PR name to reflect the driver. This is an USB IP shared between devices and not specific to a particular one.

I didn't look the driver itself and I'll review hal in another time.

BTW, make sure the driver is passing on https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/subsys/usb/testusb tests

**Pull request closed without merging**